pkg/tfbridge/defaults/unmappable.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package defaults
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Indicate that a token cannot be mapped.
+type UnmappableError struct {
+	TfToken string
+	Reason  error
+}
+
+func (err UnmappableError) Error() string {
+	return fmt.Sprintf("'%s' unmappable: %s", err.TfToken, err.Reason)
```

Comment:
It would be helpful to know if this stops tfgen or the software recovers from this error somehow, and what the user should do with this. Perhaps the idea is that the user can:

- Provide a token manually before applying defaults strategies
- Mark the resource or datasource to be skipped explicitly, thus avoiding the error

If that's the case it would be wonderful to indicate these courses of action in the error that surfaces.